experiments/compositions/composition/api/v1alpha1/getter_types.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ResourceRef struct {
+	// OPTION 1
+	// <Kind>.<group>/<version>/<namespace>/<name>
+	// resource: ServiceIdentity.serviceusage.cnrm.cloud.google.com/v1beta1//sqladmin.googleapis.com
+
+	// OPTION 2
+	Group      string `json:"group,omitempty"`
+	Version    string `json:"version,omitempty"`
+	Resource   string `json:"resource"`
+	Kind       string `json:"kind"`
+	Name       string `json:"name,omitempty"`
+	NameSuffix string `json:"nameSuffix,omitempty"`
+}
+
+type FieldRef struct {
+	Path string `json:"path"`
+	As   string `json:"as"`
+}
+
+type ValuesFrom struct {
+	Name        string      `json:"name"`
+	ResourceRef ResourceRef `json:"resourceRef"`
+	FieldRef    []FieldRef  `json:"fieldRef"`
```

Comment:
Do we need something like preconditions? (Eg status == ready?)

Reply:
Yes we do. Need to add it (separate CL ? )